es-ds-components/components/es-data-table.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+    <DataTable
+        scrollable
+        :pt="{
+            table: {
+                class: 'font-size-75 font-size-md-100',
+                style: 'border-collapse: separate; border-spacing: 0;'
+            },
+            wrapper: { 
+                style: 'border-radius: inherit'
+            },
+            thead:{
+                style: 'top: 0px; z-index: 2'
+            }
+        }"
+        :value="props.items"
+        class="border table__data-table table-spacing table"
+        :scrollHeight="scrollHeight"
+    >
+    <Column v-for="(col,index) of columns" :key="col?.key ? col.key : col" :field="col?.key ? col.key : col" :header="label(col)" 
+        :sortable="col?.sortable"
+        :frozen="checkFrozenRow(index)" alignFrozen="left"
+        :pt="{
+            sortIcon:{
+                style: 'display: none'
+            }, 
+            headerCell: (options)  => ({
+                class:[
+                    {
+                        'stickyFirstCol': options.props.frozen,
+                    }
+                ],
+            }),
+            bodyCell: (options)  => ({
+                class:[
+                    {
+                        'stickyFirstCol': options.props.frozen,
+                    }
+                ],
+            }),  
+        }"
+        >
+    </Column>
+    </DataTable>
+</template>
+<script setup lang="ts">
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+const props = defineProps({
+    fields: {
+        type: Array,
+        default: null,
+    },
+    items: {
+        type: Array,
+        required: true,
+    },
+    stickyFirstCol: {
+        type: Boolean,
+        default: false,
+    },
+    stickyHeader: {
+        type: [Boolean, String],
+        default: false,
+    },
+});
+
+const checkFrozenRow = (index) => {
+    if (index === 0 && props.stickyFirstCol) {
+        return true;
+    }
+    return false;
+};
+
+const label = (col) => {
+    if (typeof col == 'string') {
+        return convertKey(col);
+    }
+    return col?.label? col.label : convertKey(col?.key);
+};
+
+// Convert camel case and make first letter of string uppercase
+const convertKey = (string) => {
+    if (typeof string == 'string') {
+        const stringSpaced = string.replace(/([a-z])([A-Z])/g, '$1 $2');
+        return stringSpaced.charAt(0).toUpperCase() + stringSpaced.slice(1);
+    }
+    return string;  
+}
+
+const columns = computed(() => {
+    if (props.fields ) {
+        return props.fields;
+    }
+    return Object.keys(props.items[0]);
+});
+
+const scrollHeight = computed(() => {
+    if (props.stickyHeader) {
+        if(typeof props.stickyHeader == 'string'){
+            return props.stickyHeader;
+        }
+        return '300px';
+    }
+    return '';
+});
+
+</script>
+<style>
```

Comment:
Can we make sure to use `<style scoped lang="scss">` here? `.stickyFirstCol` probably needs to be wrapped in `:deep()` since it's being applied by the PrimeVue component.

Reply:
The deep comment saved me debugging time with the pointer!